src/parser/jobs/brd/modules/Sidewinder.js

```diff
@@ -0,0 +1,179 @@
+	STATUSES.STORMBITE.id,
+	STATUSES.VENOMOUS_BITE.id,
+	STATUSES.WINDBITE.id,
+]
+
+export default class Sidewinder extends Module {
+	static handle = 'sidewinder'
+	static title = t('brd.sidewinder.title')`Sidewinders and Shadowbites`
+	static dependencies = [
+		'suggestions',
+	]
+
+	_amountOfBadSidewinders = 0
+	_amountOfBadShadowbites = 0
+
+	_badCasts = []
+
+	_enemies = {}
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {
+			by: 'player',
+			abilityId: [ACTIONS.SIDEWINDER.id, ACTIONS.SHADOWBITE.id],
+		}, this._onSidewinderCast)
+
+		this.addHook('applydebuff', {
+			by: 'player',
+			abilityId: DOTS,
+		}, this._onApplyDot)
+
+		this.addHook('removedebuff', {
+			by: 'player',
+			abilityId: DOTS,
+		}, this._onRemoveDot)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onSidewinderCast(event) {
+		const enemyDots = this._getEnemyDots(event.targetID)
+		const abilityId = event.ability.guid
+		if (enemyDots.length < 2) {
+			if (abilityId === ACTIONS.SIDEWINDER.id) {
+				this._amountOfBadSidewinders++
+			} else {
+				this._amountOfBadShadowbites++
+			}
+
+			this._badCasts.push({abilityId: abilityId, dotsApplied: enemyDots.length, appliedDot: enemyDots[0], timestamp: event.timestamp})
+		}
+	}
+
+	_onApplyDot(event) {
+		const enemyDots = this._getEnemyDots(event.targetID)
+		enemyDots.push(event.ability.guid)
+	}
+
+	_onRemoveDot(event) {
+		const enemyDots = this._getEnemyDots(event.targetID)
+		const dotIndex = this._findDot(enemyDots)
+		if (dotIndex !== -1) {
+			enemyDots.splice(dotIndex, 1)
+		}
+	}
+
+	// Returns the enemy statuses state
+	_getEnemyDots(targetId) {
+		if (!this._enemies[targetId]) {
+			this._enemies[targetId] = []
+		}
+
+		return this._enemies[targetId]
+	}
+
+	_findDot(array) {
+		return array.findIndex(element => {
+			for (const dot of DOTS) {
+				if (element === dot) {
+					return true
+				}
+			}
+			return false
+		})
+	}
+
+	_onComplete() {
+		if (!this._badCasts.length) {
+			return
+		}
+		if (this._amountOfBadSidewinders) {
+			this.suggestions.add(new TieredSuggestion({
+				icon: ACTIONS.SIDEWINDER.icon,
+				content: <>
+					<Trans id="brd.sidewinder.suggestion.bad-sidewinder">
+				Only use <ActionLink {...ACTIONS.SIDEWINDER}/> when you have both <ActionLink {...ACTIONS.CAUSTIC_BITE}/> and <ActionLink {...ACTIONS.STORMBITE}/> active on the target. Remember that a DoT doesn't apply as soon as you cast it, so you have to wait for it to apply before casting <ActionLink showIcon={false} {...ACTIONS.SIDEWINDER}/>.
+					</Trans>
+				</>,
+				tiers: {
+					5: SEVERITY.MAJOR,
+					3: SEVERITY.MEDIUM,
+					1: SEVERITY.MINOR,
+				},
+				value: this._amountOfBadSidewinders,
+				why: <>
+					<Trans id="brd.sidewinder.suggestion.bad-shadowbite.reason">
+						{this._amountOfBadSidewinders} casts of <ActionLink showIcon={false} {...ACTIONS.SIDEWINDER}/> without both DoTs applied.
+					</Trans>
+				</>,
+			}))
+		}
+
+		if (this._amountOfBadShadowbites) {
```

Comment:
Probably worth combining these two suggestions, as shadowbite and sidewinder follow the same logic. There's room for a suggestion to not use shadowbite on a single target, though!